src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -94,7 +94,16 @@ public Task<Session> TryCreateServiceSessionAsync(string serviceName, Solution s
         public async Task<Session> TryCreateServiceSessionAsync(string serviceName, Solution solution, object callbackTarget, CancellationToken cancellationToken)
         {
             var snapshot = await GetPinnedScopeAsync(solution, cancellationToken).ConfigureAwait(false);
-            return await TryCreateServiceSessionAsync(serviceName, snapshot, callbackTarget, cancellationToken).ConfigureAwait(false);
+
+            try
```

Comment:
ah. yes. I see. this make the pinned scope to leak. do you know what cause TryCreateServiceSessionAsync to throw? cancellation?